### PR TITLE
Add TSB and OAB operators to Prow

### PR DESCRIPTION
**Pull request by jmontleon**

### CODE REVIEW

**Comment by djzager** [comment]
Based on `ci/prow/config':

```
{"component":"checkconfig","error":"invalid plugin configuration:\n\tplugins [trigger] are duplicated for openshift/template-service-broker-operator and openshift\n\tplugins [trigger] are duplicated for openshift/ansible-service-broker and openshift","level":"fatal","msg":"Error loading Prow plugin config.","time":"2018-11-08T16:11:15Z"}
```

It looks like the trigger plugin is already enabled? The consequence of this is not apparent.

File: ci-operator/config/openshift/ansible-service-broker/openshift-ansible-service-broker-master.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+build_root:
+  project_image_build:
```
Comment (droslean) [inline ci-operator/config/openshift/ansible-service-broker/openshift-ansible-service-broker-master.yaml]: this should be `project_image`. It seems that we have a wrong name in the validator. I've opened [202](https://github.com/openshift/ci-operator/pull/202) to fix it.

File: ci-operator/config/openshift/template-service-broker-operator/openshift-template-service-broker-operator-master.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+build_root:
+  project_image_build:
```
Comment (droslean) [inline ci-operator/config/openshift/template-service-broker-operator/openshift-template-service-broker-operator-master.yaml]: same here.

File: cluster/ci/config/prow/plugins.yaml [hunk 3]
```diff
@@ -329,6 +329,12 @@ plugins:
   openshift/cluster-autoscaler-operator:
   - approve
 
+  openshift/ansible-service-broker:
+  - trigger
```
Comment (droslean) [inline cluster/ci/config/prow/plugins.yaml]: this is already enabled in [here](https://github.com/openshift/release/blob/master/cluster/ci/config/prow/plugins.yaml#L19)

File: cluster/ci/config/prow/plugins.yaml [hunk 4]
```diff
@@ -329,6 +329,12 @@ plugins:
...
+
+  openshift/template-service-broker-operator:
+  - trigger
```
Comment (droslean) [inline cluster/ci/config/prow/plugins.yaml]: also this

File: ci-operator/config/openshift/template-service-broker-operator/openshift-template-service-broker-operator-master.yaml (hunk 2 above)
Comment (jmontleon) [reply ci-operator/config/openshift/template-service-broker-operator/openshift-template-service-broker-operator-master.yaml]: The docs here have project_image_build at https://github.com/openshift/ci-operator/blob/master/ONBOARD.md#source-code-image-target

**Comment by droslean** [comment]
You should include `tests` or `images` in the ci-operator config, depending on what you want to do :)
Also, use the `ci-operator-prowgen` to generate the prow jobs.

**Comment by jmontleon** [comment]
I'm working on figuring out the tests. I don't think we have anything written yet, so I may be working a little out of order, but we can probably do at least something to ensure the ansible being copied onto the image is valid. We also don't have anything to build as we're just adding playbooks to the ansible-operator image, so when it comes to creating image I'm not sure of the most straght forward path.

**Comment by droslean** [comment]
you can add a test to run the unitests, to start from something.

**Comment by jmontleon** [comment]
@droslean do I have any controll over the name or is this templated based on the repo name?

**Comment by droslean** [comment]
@jmontleon You should include also the prow job in this PR. You can generate is with the [prowgen](https://github.com/openshift/ci-operator-prowgen)

File: ci-operator/infra/ansible-operator-imagestream.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
```
Comment (jmontleon) [inline ci-operator/infra/ansible-operator-imagestream.yaml]: @stevekuznetsov is this file supposed to exist in ci-operator/infra or one of the sub-directories? I didn't see anything in the directory structure that was obvious to me.

**Comment by droslean** [comment]
@jmontleon You should also include those 2 repositories to the [plugins.yaml](https://github.com/openshift/release/blob/master/cluster/ci/config/prow/plugins.yaml#L170-L171) and in tide in [config.yaml](https://github.com/openshift/release/blob/master/cluster/ci/config/prow/config.yaml#L65)

**Pull request merged**